crates/red_knot_python_semantic/src/types.rs

```diff
@@ -88,9 +88,64 @@ fn symbol_ty<'db>(db: &'db dyn Db, scope: ScopeId<'db>, name: &str) -> Type<'db>
         .unwrap_or(Type::Unbound)
 }
 
-/// Shorthand for `symbol_ty` that looks up a module-global symbol by name in a file.
+/// Return a list of the symbols that typeshed declares in the scope of `types.ModuleType`
+#[salsa::tracked(return_ref)]
+fn module_type_symbols<'db>(db: &'db dyn Db) -> Box<[Box<str>]> {
```

Comment:
How many entries do we expect `module_type_symbols` to contain? Returning a list is probably fine performance wise if it is a very small number of elements. Otherwise it might be worth considering a `HashSet` (has the downside that it always requires hashing the entire input string) or a `BTreeSet`

Reply:
The query just returns all symbols declared in the body of this class definition in typeshed (excluding 3 very special attributes): https://github.com/python/typeshed/blob/c225ac758781c0599889b320c0a4f334366c5907/stdlib/types.pyi#L331. Currently that's a list of length 6; it might change in the future (which is why it's a salsa query rather than a hardcoded list) but I do not expect it to ever be very long. For this reason I did not go for a set, as I assumed that the cost of hashing the strings would be more costly than it was worth.